### PR TITLE
Dragon fix

### LINKED ISSUE #960: [Steam Version] Game freezes when any Krumblor upgrade is bought
**Describe the bug**
Purchasing the "Train Breath of Milk" Krumblor upgrade causes the game to freeze just like issue #946. (Tried loading up CM again with other Krumblor upgrades and the issue still persists.)
Any buildings, upgrades, or even the big cookie will not respond when clicked, game progress is also frozen.
However, it is still possible to navigate to and through the options menu to disable CM. 
Doesn't freeze after disabling CM and restarting the game.

**To Reproduce**
1. Load the steam version of the game with Cookie Monster enabled.
2. Buy any Krumblor upgrade.

**Screenshots**

**Save file**
[Poo.txt](https://github.com/CookieMonsterTeam/CookieMonster/files/7104341/Poo.txt)

**Browser**
Steam version.

### PR DESCRIPTION
**Pull request by Urganot**
This closes #960

### CODE REVIEW

**Comment by DanielNoord** [comment]
I think you might need to rebuild CookieMonster.js after merging the dev branch. It complains about a difference between the current file in your branch and a temporary file created by running the build command again (ie., if you run the build command on your branch now you get a different file than is currently included on the branch).

**Comment by DanielNoord** [comment]
Looks good! Going to test this when I get home, but LGTM!

**Comment by Urganot** [comment]
I just realised something. In the last commit there is a change to the mana bar text because in steam it will print it 5 times.
this will more or less diable the timer in the web version

**Comment by DanielNoord** [comment]
Are you able to also fix that?

**Comment by Urganot** [comment]
Without having a seperate version or checking the version number in the script, I dont know how

**Comment by DanielNoord** [comment]
Doesn't this work?
```javascript
if (typeof Steam !== 'undefined') {
 // Steam function
} else {
 // Function for browser version
}
```

File: src/Main/ReplaceGameElements/NativeGrimoire.js [hunk 1]
```diff
@@ -13,7 +13,7 @@ import ReplaceTooltipGrimoire from './TooltipGrimoire';
  * This function fixes replaces the .draw function of the Grimoire
  */
 function ReplaceNativeGrimoireDraw() {
-  if (!HasReplaceNativeGrimoireDraw && Game.Objects['Wizard tower'].minigameLoaded) {
+  if (Game.drawT % 5 === 0 && !HasReplaceNativeGrimoireDraw && Game.Objects['Wizard tower'].minigameLoaded) {
```
Comment (DanielNoord) [inline src/Main/ReplaceGameElements/NativeGrimoire.js:16]: Why is this needed?

**Comment by dmytrokyrychuk** [comment]
Isn't this simply going to calculate the price of all the buildings that have at least `amount` owned, rather than the cost of the bulding required for the current dragon level upgrade? Am I reading the code correctly?

See my attempt at #975

File: src/Main/ReplaceGameElements/NativeGrimoire.js (hunk 1 above)
Comment (Urganot) [reply src/Main/ReplaceGameElements/NativeGrimoire.js:16]: The game now draws the mana bar text every 5 frames instead of every frame. without this change, CM will append the timer 5 times before the game resets it

File: package.json [hunk 2]
```diff
@@ -46,21 +46,21 @@
   "devDependencies": {
     "@types/chai": "^4.2.19",
     "@types/mocha": "^9.0.0",
-    "@typescript-eslint/eslint-plugin": "^4.30.0",
-    "@typescript-eslint/parser": "^4.30.0",
+    "@typescript-eslint/eslint-plugin": "^4.29.1",
```
Comment (DanielNoord) [inline package.json:49]: Why did you downgrade this?

File: src/Main/ReplaceGameElements/NativeGrimoire.js (hunk 1 above)
Comment (DanielNoord) [reply src/Main/ReplaceGameElements/NativeGrimoire.js:16]: This has been fixed in another PR! See #976

**Comment by DanielNoord** [comment]
Thanks for your help guys, with #981 this has now been fixed!

File: src/Cache/Dragon/Dragon.js [hunk 3]
```diff
@@ -14,51 +14,30 @@ export default function CacheDragonCost() {
...
-                Math.max(0, SimObjects[target].amount - 1 - SimObjects[target].free);
-            price = Game.modifyBuildingPrice(SimObjects[target], price);
-            price = Math.ceil(price);
-            cost += price;
-            SimObjects[target].amount -= 1;
-          }
-          CacheCostDragonUpgrade = `Cost to rebuy: ${Beautify(cost)}`;
-        });
-      }
+        for (let i = 0; i < amount; i++) {
```
Comment (lclauser) [inline src/Cache/Dragon/Dragon.js:29]: Sorry to intrude but why don't we use this function ([BuyBuilding](https://github.com/CookieMonsterTeam/CookieMonster/blob/195d7335f8be7d7f259a4c8cec48eea3abf4e92c/src/Sim/SimulationEvents/BuyBuilding.js)) instead of calculating it here?
Something like 
```js
BuildingGetPrice(
    target,
    SimObjects[target].basePrice,
    SimObjects[target].amount - amount,
    SimObjects[target].free,
    amount)
```
